### PR TITLE
ath79: Add support for Archer C60 v1/v2

### PR DESCRIPTION
**Pull request by adschm**
This adds support for the Archer C60 v1/v2.

~~I currently only own the v1 and tested on this one. If someone wants to test on v2, I would be happy. Otherwise, the v2 patch will be stashed until I can get one myself (not so soon).~~

~~This is RFC, everything works EXCEPT the network ports. Setup looks okay, but no communication seems to be possible. Hints are welcome.~~
Edit: C60 v1 successfully tested.

Update: C60 v1 and v2 successfully tested. Ready for merge.

### CODE REVIEW

File: target/linux/ath79/base-files/etc/board.d/01_leds [hunk 1]
```diff
@@ -116,6 +116,7 @@ tplink,tl-wr1043nd-v4)
 	;;
 tplink,archer-c58-v1|\
 tplink,archer-c59-v1|\
+tplink,archer-c60-v1|\
 tplink,archer-c60-v1)
```
[the remark this answers is not in the document]
Comment (adrianschmutzler) [reply target/linux/ath79/base-files/etc/board.d/01_leds]: Yes, indeed.

**Comment by adrianschmutzler** [comment]
Looks like I've messed up my client computers ethernet the other day.
Good news: Ethernet on the C60 seems to be working. Nevertheless, I will have another look on this.

**Comment by adrianschmutzler** [comment]
Successfully tested all ports on C60 v1.

Bootlog:
```
[    0.000000] Linux version 4.14.113 (adsc@buildfff) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r9915-d6643aca34)) #0 Wed May 1 05:25:55 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is TP-Link Archer C60 v1
[    0.000000] SoC: Qualcomm Atheros QCA956X ver 1 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000] free_area_init_node: node 0, pgdat 80501c90, node_mem_map 81007e60
[    0.000000]   Normal zone: 128 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x8c/0x470 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16256
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] PID hash table entries: 256 (order: -2, 1024 bytes)
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 58340K/65536K available (3783K kernel code, 152K rwdata, 812K rodata, 1196K init, 204K bss, 7196K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 775.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 4932285024 ns
[    0.000007] sched_clock: 32 bits at 387MHz, resolution 2ns, wraps every 5541893118ns
[    0.008213] Calibrating delay loop... 385.84 BogoMIPS (lpj=1929216)
[    0.074760] pid_max: default: 32768 minimum: 301
[    0.079776] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086756] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.097614] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.108004] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.114496] pinctrl core: initialized pinctrl subsystem
[    0.120816] NET: Registered protocol family 16
[    0.131217] PCI host bridge /ahb/apb/pcie-controller@18250000 ranges:
[    0.138017]  MEM 0x0000000012000000..0x0000000013ffffff
[    0.143550]   IO 0x0000000000000000..0x0000000000000000
[    0.163894] PCI host bridge to bus 0000:00
[    0.168216] pci_bus 0000:00: root bus resource [mem 0x12000000-0x13ffffff]
[    0.175488] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.181355] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.188512] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.196884] pci 0000:00:00.0: [168c:0056] type 00 class 0x028000
[    0.196939] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.197059] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.197231] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.197252] pci 0000:00:00.0: BAR 0: assigned [mem 0x12000000-0x121fffff 64bit]
[    0.207200] clocksource: Switched to clocksource MIPS
[    0.213400] NET: Registered protocol family 2
[    0.218679] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.226033] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.232780] TCP: Hash tables configured (established 1024 bind 1024)
[    0.239613] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.245781] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.252629] NET: Registered protocol family 1
[    0.257281] PCI: CLS 0 bytes, default 32
[    0.259752] Crashlog allocated RAM at address 0x3f00000
[    0.266335] workingset: timestamp_bits=30 max_order=14 bucket_order=0
[    0.277160] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.283345] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.301142] io scheduler noop registered
[    0.305278] io scheduler deadline registered (default)
[    0.312382] pinctrl-single 1804002c.pinmux: 544 pins at pa b804002c size 68
[    0.320686] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.328036] console [ttyS0] disabled
[    0.331837] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 1562500) is a 16550A
[    0.340966] console [ttyS0] enabled
[    0.348521] bootconsole [early0] disabled
[    0.364281] m25p80 spi0.0: gd25q64 (8192 Kbytes)
[    0.369139] 5 fixed-partitions partitions found on MTD device spi0.0
[    0.375692] Creating 5 MTD partitions on "spi0.0":
[    0.380662] 0x000000000000-0x000000010000 : "u-boot"
[    0.386480] 0x000000010000-0x000000020000 : "mac"
[    0.392044] 0x000000020000-0x0000007e0000 : "firmware"
[    0.400220] 2 uimage-fw partitions found on MTD device firmware
[    0.406338] Creating 2 MTD partitions on "firmware":
[    0.411509] 0x000000000000-0x000000184c35 : "kernel"
[    0.417300] 0x000000184c35-0x0000007c0000 : "rootfs"
[    0.423043] mtd: device 4 (rootfs) set to be root filesystem
[    0.428981] 1 squashfs-split partitions found on MTD device rootfs
[    0.435368] 0x0000004b0000-0x0000007c0000 : "rootfs_data"
[    0.441584] 0x0000007e0000-0x0000007f0000 : "tplink"
[    0.447386] 0x0000007f0000-0x000000800000 : "art"
[    0.453851] libphy: Fixed MDIO Bus: probed
[    0.797560] libphy: ag71xx_mdio: probed
[    0.802197] libphy: ar8xxx-mdio: probed
[    0.809969] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.222847] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:00 [uid=004dd042, driver=Generic PHY]
[    1.233202] eth0: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    1.578802] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.588593] eth1: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.596255] GACT probability on
[    1.599553] u32 classifier
[    1.602342]     input device check on
[    1.606113]     Actions configured
[    1.610439] NET: Registered protocol family 10
[    1.619370] Segment Routing with IPv6
[    1.623243] NET: Registered protocol family 17
[    1.627917] 8021q: 802.1Q VLAN Support v1.8
[    1.639027] VFS: Mounted root (squashfs filesystem) readonly on device 31:4.
[    1.652831] Freeing unused kernel memory: 1196K
[    1.657533] This architecture does not have kernel memory protection.
[    2.432917] init: Console is alive
[    2.436612] init: - watchdog -
[    2.450279] random: fast init done
[    3.754790] kmodloader: loading kernel modules from /etc/modules-boot.d/*
[    4.073350] usbcore: registered new interface driver usbfs
[    4.079164] usbcore: registered new interface driver hub
[    4.084729] usbcore: registered new device driver usb
[    4.098524] ehci_hcd: USB 2.0 'Enhanced' Host Controller (EHCI) Driver
[    4.106678] ehci-platform: EHCI generic platform driver
[    4.114605] ohci_hcd: USB 1.1 'Open' Host Controller (OHCI) Driver
[    4.122239] ohci-platform: OHCI generic platform driver
[    4.128404] kmodloader: done loading kernel modules from /etc/modules-boot.d/*
[    4.146256] init: - preinit -
[    4.868335] random: jshn: uninitialized urandom read (4 bytes read)
[    5.163298] random: jshn: uninitialized urandom read (4 bytes read)
[    5.302353] random: jshn: uninitialized urandom read (4 bytes read)
[    6.361516] IPv6: ADDRCONF(NETDEV_UP): eth1: link is not ready
[    6.367659] IPv6: ADDRCONF(NETDEV_UP): eth1.1: link is not ready
[    6.430587] urandom_read: 4 callbacks suppressed
[    6.430594] random: procd: uninitialized urandom read (4 bytes read)
[    7.388371] eth1: link up (1000Mbps/Full duplex)
[    7.393169] IPv6: ADDRCONF(NETDEV_CHANGE): eth1: link becomes ready
[    7.399962] IPv6: ADDRCONF(NETDEV_CHANGE): eth1.1: link becomes ready
[    9.719786] jffs2_scan_eraseblock(): End of filesystem marker found at 0x10000
[    9.727270] jffs2_build_filesystem(): unlocking the mtd device...
[    9.727312] done.
[    9.735675] jffs2_build_filesystem(): erasing all blocks after the end marker...
[   22.699150] done.
[   22.708906] jffs2: notice: (461) jffs2_build_xattr_subsystem: complete building xattr subsystem, 0 of xdatum (0 unchecked, 0 orphan) and 0 of xref (0 dead, 0 orphan) found.
[   22.726134] mount_root: overlay filesystem has not been fully initialized yet
[   22.735559] mount_root: switching to jffs2 overlay
[   22.782530] overlayfs: upper fs does not support tmpfile.
[   22.838969] urandom-seed: Seed file not found (/etc/urandom.seed)
[   22.872787] eth1: link down
[   22.891499] procd: - early -
[   22.894560] procd: - watchdog -
[   23.447161] procd: - watchdog -
[   23.450688] procd: - ubus -
[   23.504248] random: ubusd: uninitialized urandom read (4 bytes read)
[   23.511246] random: ubusd: uninitialized urandom read (4 bytes read)
[   23.518630] procd: - init -
[   23.724790] kmodloader: loading kernel modules from /etc/modules.d/*
```

**Comment by adrianschmutzler** [comment]
@heinzek
Since you added support for C60v2 in ar71xx, maybe you want to have a look at this. Note that v2 in this PR is completely untested.

**Comment by adrianschmutzler** [comment]
@hmh 
In case you are interested in initial testing.

**Comment by hmh** [comment]
Yikes, wasn't ath79 supposed to be *smaller* than ar71xx?   Anyway, I am trying, but so far I failed to get any image generated because something inflated them when going from ar71xx -> ath79.

**Comment by adrianschmutzler** [comment]
WAN was linked to swphy0 instead of swphy4. Fixed that and retested all ports on C60v1.

**Comment by adrianschmutzler** [comment]
This will benefit from some suggested changes for C25 in https://github.com/openwrt/openwrt/pull/2056

This interferes with https://github.com/openwrt/openwrt/pull/2066
I'm planning to update this one to rebase onto 2066 ...

**Comment by adrianschmutzler** [comment]
@hmh 
I build with a special setup/configuration, but all my images from master (sysupgrade) have 4801 kiB. (v1/v2, ar71xx/ath79).
openwrt-18.06 with the same setup was 4609 kiB.
Unfortunately, I cannot help you much beyond "it should work" ...

**Comment by adrianschmutzler** [comment]
This is a reminder to adapt the mib-poll-interval: 6648e9458f2f47cdbe16bd9537f08048a0a252fd

**Comment by lynxis** [comment]
@adrianschmutzler is this still an RFC or ready to merge upstream?

**Comment by adrianschmutzler** [comment]
@lynxis
The v1 is tested and working, but I would update it slightly as mentioned and then retest myself.
V2 is completely untested, since I have no device. I just provided this one as it is straightforward.
However, technically it is not RFC anymore, but RFT.

I will repost soon when I've updated.

**Comment by adrianschmutzler** [comment]
Okay, so I updated SoC and gpio-leds compatible, rebased/adjusted on top of PR https://github.com/openwrt/openwrt/pull/2066, and retested V1 successfully.

**Comment by adrianschmutzler** [comment]
Rebased onto master with eth0/eth1 swap patch merged.

**Comment by adrianschmutzler** [comment]
I finally obtained a C60 v2 myself and successfully tested.

@lynxis RFC can be removed, this should be ready for merge now.

Bootlog from V2:

```
[    0.000000] Linux version 4.19.53 (adsc@buildfff) (gcc version 7.4.0 (OpenWrt GCC 7.4.0 r10291-5470e928d0)) #0 Thu Jun 20 18:02:30 2019
[    0.000000] bootconsole [early0] enabled
[    0.000000] CPU0 revision is: 00019750 (MIPS 74Kc)
[    0.000000] MIPS: machine is TP-Link Archer C60 v2
[    0.000000] SoC: Qualcomm Atheros QCA956X ver 1 rev 0
[    0.000000] Determined physical RAM map:
[    0.000000]  memory: 04000000 @ 00000000 (usable)
[    0.000000] Initrd not found or empty - disabling initrd
[    0.000000] Primary instruction cache 64kB, VIPT, 4-way, linesize 32 bytes.
[    0.000000] Primary data cache 32kB, 4-way, VIPT, cache aliases, linesize 32 bytes
[    0.000000] Zone ranges:
[    0.000000]   Normal   [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Movable zone start for each node
[    0.000000] Early memory node ranges
[    0.000000]   node   0: [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] Initmem setup node 0 [mem 0x0000000000000000-0x0000000003ffffff]
[    0.000000] On node 0 totalpages: 16384
[    0.000000]   Normal zone: 144 pages used for memmap
[    0.000000]   Normal zone: 0 pages reserved
[    0.000000]   Normal zone: 16384 pages, LIFO batch:3
[    0.000000] random: get_random_bytes called from start_kernel+0x98/0x4a4 with crng_init=0
[    0.000000] pcpu-alloc: s0 r0 d32768 u32768 alloc=1*32768
[    0.000000] pcpu-alloc: [0] 0
[    0.000000] Built 1 zonelists, mobility grouping on.  Total pages: 16240
[    0.000000] Kernel command line: console=ttyS0,115200n8 rootfstype=squashfs,jffs2
[    0.000000] Dentry cache hash table entries: 8192 (order: 3, 32768 bytes)
[    0.000000] Inode-cache hash table entries: 4096 (order: 2, 16384 bytes)
[    0.000000] Writing ErrCtl register=00000000
[    0.000000] Readback ErrCtl register=00000000
[    0.000000] Memory: 57516K/65536K available (4304K kernel code, 168K rwdata, 1016K rodata, 1224K init, 207K bss, 8020K reserved, 0K cma-reserved)
[    0.000000] SLUB: HWalign=32, Order=0-3, MinObjects=0, CPUs=1, Nodes=1
[    0.000000] NR_IRQS: 51
[    0.000000] CPU clock: 775.000 MHz
[    0.000000] clocksource: MIPS: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 4932285024 ns
[    0.000007] sched_clock: 32 bits at 387MHz, resolution 2ns, wraps every 5541893118ns
[    0.008216] Calibrating delay loop... 385.84 BogoMIPS (lpj=1929216)
[    0.074762] pid_max: default: 32768 minimum: 301
[    0.079804] Mount-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.086795] Mountpoint-cache hash table entries: 1024 (order: 0, 4096 bytes)
[    0.098382] clocksource: jiffies: mask: 0xffffffff max_cycles: 0xffffffff, max_idle_ns: 19112604462750000 ns
[    0.108787] futex hash table entries: 256 (order: -1, 3072 bytes)
[    0.115334] pinctrl core: initialized pinctrl subsystem
[    0.121702] NET: Registered protocol family 16
[    0.132722] PCI host bridge /ahb/apb/pcie-controller@18250000 ranges:
[    0.139525]  MEM 0x0000000012000000..0x0000000013ffffff
[    0.145073]   IO 0x0000000000000000..0x0000000000000000
[    0.168295] PCI host bridge to bus 0000:00
[    0.172618] pci_bus 0000:00: root bus resource [mem 0x12000000-0x13ffffff]
[    0.179912] pci_bus 0000:00: root bus resource [io  0x0000]
[    0.185781] pci_bus 0000:00: root bus resource [??? 0x00000000 flags 0x0]
[    0.192941] pci_bus 0000:00: No busn resource found for root bus, will use [bus 00-ff]
[    0.201318] pci 0000:00:00.0: [168c:0056] type 00 class 0x028000
[    0.201373] pci 0000:00:00.0: reg 0x10: [mem 0x00000000-0x001fffff 64bit]
[    0.201495] pci 0000:00:00.0: PME# supported from D0 D3hot
[    0.202397] pci_bus 0000:00: busn_res: [bus 00-ff] end is updated to 00
[    0.202423] pci 0000:00:00.0: BAR 0: assigned [mem 0x12000000-0x121fffff 64bit]
[    0.213090] clocksource: Switched to clocksource MIPS
[    0.219425] NET: Registered protocol family 2
[    0.224813] tcp_listen_portaddr_hash hash table entries: 512 (order: 0, 4096 bytes)
[    0.232929] TCP established hash table entries: 1024 (order: 0, 4096 bytes)
[    0.240325] TCP bind hash table entries: 1024 (order: 0, 4096 bytes)
[    0.247045] TCP: Hash tables configured (established 1024 bind 1024)
[    0.253876] UDP hash table entries: 256 (order: 0, 4096 bytes)
[    0.260044] UDP-Lite hash table entries: 256 (order: 0, 4096 bytes)
[    0.266996] NET: Registered protocol family 1
[    0.271628] PCI: CLS 0 bytes, default 32
[    0.274345] Crashlog allocated RAM at address 0x3f00000
[    0.281099] workingset: timestamp_bits=14 max_order=14 bucket_order=0
[    0.293700] squashfs: version 4.0 (2009/01/31) Phillip Lougher
[    0.299845] jffs2: version 2.2 (NAND) (SUMMARY) (LZMA) (RTIME) (CMODE_PRIORITY) (c) 2001-2006 Red Hat, Inc.
[    0.322227] io scheduler noop registered
[    0.326407] io scheduler deadline registered (default)
[    0.333570] pinctrl-single 1804002c.pinmux: 544 pins, size 68
[    0.340560] Serial: 8250/16550 driver, 1 ports, IRQ sharing disabled
[    0.347927] console [ttyS0] disabled
[    0.351721] 18020000.uart: ttyS0 at MMIO 0x18020000 (irq = 9, base_baud = 1562500) is a 16550A
[    0.360859] console [ttyS0] enabled
[    0.368416] bootconsole [early0] disabled
[    0.384779] m25p80 spi0.0: gd25q64 (8192 Kbytes)
[    0.389598] 6 fixed-partitions partitions found on MTD device spi0.0
[    0.396184] Creating 6 MTD partitions on "spi0.0":
[    0.401151] 0x000000000000-0x00000001fb00 : "factory-boot"
[    0.407535] 0x00000001fb00-0x000000020000 : "mac"
[    0.413179] 0x000000020000-0x000000030000 : "u-boot"
[    0.418974] 0x000000030000-0x0000007d0000 : "firmware"
[    0.428143] 2 uimage-fw partitions found on MTD device firmware
[    0.434303] Creating 2 MTD partitions on "firmware":
[    0.439444] 0x000000000000-0x0000001c0000 : "kernel"
[    0.445246] 0x0000001c0000-0x0000007a0000 : "rootfs"
[    0.450983] mtd: device 5 (rootfs) set to be root filesystem
[    0.458213] 1 squashfs-split partitions found on MTD device rootfs
[    0.464653] 0x0000004d0000-0x0000007a0000 : "rootfs_data"
[    0.470943] 0x0000007d0000-0x0000007f0000 : "tplink"
[    0.476804] 0x0000007f0000-0x000000800000 : "art"
[    0.483347] libphy: Fixed MDIO Bus: probed
[    0.834172] ag71xx 19000000.eth: Could not connect to PHY device. Deferring probe.
[    1.513480] libphy: ag71xx_mdio: probed
[    1.518115] libphy: ar8xxx-mdio: probed
[    1.526093] switch0: Atheros AR8229 rev. 1 switch registered on mdio-bus.0
[    1.598541] ag71xx 1a000000.eth: connected to PHY at fixed-0:00 [uid=00000000, driver=Generic PHY]
[    1.608372] eth0: Atheros AG71xx at 0xba000000, irq 5, mode: gmii
[    1.616189] GACT probability on
[    1.619447] u32 classifier
[    1.622234]     input device check on
[    1.626055]     Actions configured
[    1.630516] NET: Registered protocol family 10
[    1.640309] Segment Routing with IPv6
[    1.644242] NET: Registered protocol family 17
[    1.648901] 8021q: 802.1Q VLAN Support v1.8
[    1.998747] ag71xx 19000000.eth: connected to PHY at mdio-bus.0:1f:04 [uid=004dd042, driver=Generic PHY]
[    2.009417] eth1: Atheros AG71xx at 0xb9000000, irq 4, mode: mii
[    2.024098] VFS: Mounted root (squashfs filesystem) readonly on device 31:5.
[    2.038070] Freeing unused kernel memory: 1224K
[    2.042748] This architecture does not have kernel memory protection.
[    2.049423] Run /sbin/init as init process
[    2.453101] random: fast init done
```

**Comment by adrianschmutzler** [comment]
rebased and added label-mac-device

**Comment by ynezz** [comment]
Please add flashing instructions to your commit messages.

File: target/linux/ath79/dts/qca9561_tplink_archer-c6x.dtsi [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+
+&pcie {
+	status = "okay";
+};
+
+&eth0 {
+	phy-mode = "mii";
+	phy-handle = <&swphy4>;
+
+	status = "okay";
```
Comment (ynezz) [inline target/linux/ath79/dts/qca9561_tplink_archer-c6x.dtsi]: Just a small nitpick, there's a convention to start with a `status` property, then usually newline followed by the rest of the properties.

File: target/linux/ath79/base-files/etc/board.d/01_leds (hunk 1 above)
[the remark this answers is not in the document]
Comment (ynezz) [reply target/linux/ath79/base-files/etc/board.d/01_leds]: So? :-)

[the remark this answers is not in the document]
Comment (ynezz) [reply target/linux/ath79/base-files/etc/board.d/01_leds]: Ah, I see now.

**Comment by adrianschmutzler** [comment]
Update:
- added flashing instructions (copied with some adjustments from ar71xx)
- aligned `status = ...` in DTS

**Comment by ynezz** [comment]
Thanks! Pulled into my staging tree at https://git.openwrt.org/openwrt/staging/ynezz.git

**Comment by castillofrancodamian** [comment]
It has many mistakes. WiFi is disconnected and unstable. I have version 1.

**Pull request closed without merging**